homeassistant/components/sun/__init__.py

```diff
@@ -97,15 +98,19 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     # we will create entities before firing EVENT_COMPONENT_LOADED
     await async_process_integration_platform_for_component(hass, DOMAIN)
     hass.data[DOMAIN] = Sun(hass)
+    await hass.config_entries.async_forward_entry_setups(entry, [Platform.SENSOR])
     return True
 
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
-    sun = hass.data.pop(DOMAIN)
-    sun.remove_listeners()
-    hass.states.async_remove(sun.entity_id)
-    return True
+    if unload_ok := await hass.config_entries.async_unload_platforms(
```

Comment:
A note: What minimum python version does home assistant support?

The walrus operator is only supported in python 3.8+, while [python 3.7 is not yet EOL](https://endoflife.date/python).

Reply:
3.9

Reply:
👍

apologies for the noise, /res